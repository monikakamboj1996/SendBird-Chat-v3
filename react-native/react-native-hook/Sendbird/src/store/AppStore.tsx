import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
