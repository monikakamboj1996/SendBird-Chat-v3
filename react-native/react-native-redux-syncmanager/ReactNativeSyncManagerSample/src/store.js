import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
