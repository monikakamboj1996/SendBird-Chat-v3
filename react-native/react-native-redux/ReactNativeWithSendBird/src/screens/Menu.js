import React, { Component } from 'react';
import { View, Platform } from 'react-native';
import AsyncStorage from '@react-native-community/async-storage';
import { connect } from 'react-redux';
import firebase from 'react-native-firebase';
import { sendbirdLogout, initMenu } from '../actions';
import { sbConnect, sbRegisterPushToken, sbUnregisterPushToken } from '../sendbirdActions';
import { NavigationActions, StackActions } from 'react-navigation';
import { Button, HR, Spinner } from '../components';

class Menu extends Component {
  static navigationOptions = {
    title: 'Sendbird'
  };

  constructor(props) {
    super(props);
    this.pushNotificationEnabled = false;
    this.state = {
      isLoading: false
    };
  }

  componentDidMount() {
    this.onTokenRefreshListener = firebase.messaging().onTokenRefresh(fcmToken => {
      sbRegisterPushToken()
        .then(res => {})
        .catch(err => {});
    });
    this.onMessageListener = firebase.messaging().onMessage(message => {
      if (Platform.OS === 'ios') {
        const text = message.data.message;
        const payload = JSON.parse(message.data.sendbird);
        const localNotification = new firebase.notifications.Notification({
          show_in_foreground: true
        })
          .setNotificationId(message.messageId)
          .setTitle('New message')
          .setSubtitle(`Unread message: ${payload.unread_message_count}`)
          .setBody(text)
          .setData(payload);
        firebase.notifications().displayNotification(localNotification);
      }
    });
    this.onPushNotificationOpened = firebase.notifications().onNotificationOpened(notif => {
      firebase.notifications().removeAllDeliveredNotifications();
      AsyncStorage.getItem('user', (err, result) => {
        if (result) {
          const user = JSON.parse(result);
          sbConnect(user.userId, user.nickname)
            .then(() => {
              let payload = notif.notification.data;
              if (!payload.hasOwnProperty('channel') && payload.hasOwnProperty('sendbird')) {
                payload = payload.sendbird;
              }
              const isOpenChannel = () => {
                return payload.channel_type !== 'group_messaging' && payload.channel_type !== 'messaging';
              };
              const channelType = isOpenChannel() ? 'OpenChannel' : 'GroupChannel';
              this.props.navigation.dispatch(
                StackActions.reset({
                  index: 2,
                  actions: [
                    NavigationActions.navigate({ routeName: 'Menu' }),
                    NavigationActions.navigate({ routeName: channelType }),
                    NavigationActions.navigate({
                      routeName: 'Chat',
                      params: {
                        channelUrl: payload.channel.channel_url,
                        newTitle: payload.channel.name,
                        isOpenChannel: isOpenChannel(),
                        isFromPayload: true
                      }
                    })
                  ]
                })
              );
            })
            .catch(err => {
              this.redirectTo('Login');
            });
        } else this.redirectTo('Login');
      });
    });
    firebase
      .messaging()
      .hasPermission()
      .then(enabled => {
        if (enabled) {
          this.pushNotificationEnabled = true;
        } else {
          firebase
            .messaging()
            .requestPermission()
            .then(() => {
              this.pushNotificationEnabled = true;
            })
            .catch(error => {
              this.pushNotificationEnabled = false;
            });
        }
      });
    firebase.notifications().removeAllDeliveredNotifications();
    this.props.initMenu();
  }
  componentWillUnmount() {
    this.onTokenRefreshListener();
    this.onMessageListener();
    this.onPushNotificationOpened();
  }

  componentDidUpdate(prevProps) {
    AsyncStorage.getItem('user', (err, result) => {
      if (!result) {
        this.setState({ isLoading: false }, () => {
          const resetAction = StackActions.reset({
            index: 0,
            actions: [NavigationActions.navigate({ routeName: 'Login' })]
          });
          this.props.navigation.dispatch(resetAction);
        });
      }
    });
  }

  redirectTo(page, params) {
    this.props.navigation.dispatch(
      StackActions.reset({
        index: 0,
        actions: [NavigationActions.navigate({ routeName: page, params: params })]
      })
    );
  }

  _onProfileButtonPress = () => {
    this.props.navigation.navigate('Profile');
  };

  _onOpenChannelPress = () => {
    this.props.navigation.navigate('OpenChannel');
  };

  _onGroupChannelPress = () => {
    this.props.navigation.navigate('GroupChannel');
  };

  _onDisconnectButtonPress = () => {
    this.setState({ isLoading: true }, () => {
      sbUnregisterPushToken()
        .then(res => {
          this.props.sendbirdLogout();
        })
        .catch(err => {
          this.setState({ isLoading: false });
        });
    });
  };

  render() {
    return (
      <View style={styles.containerViewStyle}>
        <Spinner visible={this.state.isLoading} />
        <Button
          containerViewStyle={styles.menuViewStyle}
          buttonStyle={styles.buttonStyle}
          backgroundColor="#fff"
          color="#6e5baa"
          icon={{
            name: 'user',
            type: 'font-awesome',
            color: '#6e5baa',
            size: 16
          }}
          title="Profile"
          onPress={this._onProfileButtonPress}
        />
        <HR />
        <Button
          containerViewStyle={styles.menuViewStyle}
          buttonStyle={styles.buttonStyle}
          backgroundColor="#fff"
          color="#6e5baa"
          icon={{
            name: 'slack',
            type: 'font-awesome',
            color: '#6e5baa',
            size: 16
          }}
          title="Open Channel"
          onPress={this._onOpenChannelPress}
        />
        <HR />
        <Button
          containerViewStyle={styles.menuViewStyle}
          buttonStyle={styles.buttonStyle}
          backgroundColor="#fff"
          color="#6e5baa"
          icon={{
            name: 'users',
            type: 'font-awesome',
            color: '#6e5baa',
            size: 16
          }}
          title="Group Channel"
          onPress={this._onGroupChannelPress}
        />
        <HR />
        <Button
          containerViewStyle={styles.menuViewStyle}
          buttonStyle={styles.buttonStyle}
          backgroundColor="#fff"
          color="#7d62d9"
          color="#6e5baa"
          icon={{
            name: 'sign-out',
            type: 'font-awesome',
            color: '#6e5baa',
            size: 16
          }}
          title="Disconnect"
          onPress={this._onDisconnectButtonPress}
        />
        <HR />
      </View>
    );
  }
}

function mapStateToProps({ menu }) {
  const { isDisconnected } = menu;
  return { isDisconnected };
}

export default connect(
  mapStateToProps,
  { sendbirdLogout, initMenu }
)(Menu);

const styles = {
  containerViewStyle: {
    backgroundColor: '#fff',
    flex: 1
  },
  menuViewStyle: {
    marginLeft: 0,
    marginRight: 0
  },
  buttonStyle: {
    justifyContent: 'flex-start',
    paddingLeft: 14
  }
};
