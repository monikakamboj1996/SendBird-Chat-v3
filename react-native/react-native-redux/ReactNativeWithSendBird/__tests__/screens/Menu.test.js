import 'react-native';
import React from 'react';
import Menu from '../../src/screens/Menu';
import store from '../../src/store';
import renderer from 'react-test-renderer';

describe('screen/Menu', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Menu store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
