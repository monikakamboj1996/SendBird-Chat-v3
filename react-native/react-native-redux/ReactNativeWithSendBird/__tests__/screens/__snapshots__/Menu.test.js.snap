// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/Menu renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "justifyContent": "flex-start",
              "paddingLeft": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#6e5baa",
                "fontSize": 16,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="font-awesome"
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                Object {
                  "color": "#6e5baa",
                },
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Profile
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "flexDirection": "row",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "justifyContent": "flex-start",
              "paddingLeft": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#6e5baa",
                "fontSize": 16,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="font-awesome"
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                Object {
                  "color": "#6e5baa",
                },
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Open Channel
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "flexDirection": "row",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "justifyContent": "flex-start",
              "paddingLeft": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#6e5baa",
                "fontSize": 16,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="font-awesome"
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                Object {
                  "color": "#6e5baa",
                },
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Group Channel
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "flexDirection": "row",
          "height": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        Object {
          "marginLeft": 0,
          "marginRight": 0,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#fff",
            },
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            Object {
              "justifyContent": "flex-start",
              "paddingLeft": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#6e5baa",
                "fontSize": 16,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                undefined,
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="font-awesome"
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                Object {
                  "color": "#6e5baa",
                },
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Disconnect
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e6e6e6",
          "flexDirection": "row",
          "height": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
